### PR TITLE
S9 Review Overview

### PR DESCRIPTION
**Pull request by DarkPurple141**
Technically this solves the title, but many other changes:

1. fixed a bug with votes
2. did some minor refactoring
3. adding a new component to manage recommend

### CODE REVIEW

File: frontend/src/components/Vote/base.vue [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <template>
     <div class="meta-fields">
-        <p v-on:click="upvote" class="material-icons vote">keyboard_arrow_up</p>
+        <p v-on:click="upvote" :class="[userLiked === 1 ? 'voted' : '', 'material-icons vote']">keyboard_arrow_up</p>
         <p class="likes">{{ likes }}</p>
-        <p v-on:click="downvote" class="material-icons vote">keyboard_arrow_down</p>
+        <p v-on:click="downvote" :class="[userLiked === -1 ? 'voted' : '','material-icons vote']">keyboard_arrow_down</p>
```
Comment (PapaPwn) [inline frontend/src/components/Vote/base.vue]: It took me a little while to actually notice the colour change of the arrows... Can we make a more noticeable change?

File: backend/src/controllers/course.js [hunk 2]
```diff
@@ -40,12 +40,12 @@ exports.getCourseQuestions = function ({ params, query }, res) {
             }
             const lastPage = Math.trunc((questionCount[0]['COUNT()'] + pageSize - 1) / pageSize)
             return {
-                'meta': {
-                    'curr': pageNumber,
-                    'last': lastPage || 1,
-                    'pageSize': pageSize
+                meta: {
```
Comment (NunoDasNeves) [inline backend/src/controllers/course.js]: why :(

File: frontend/src/components/Vote/base.vue (hunk 1 above)
Comment (NunoDasNeves) [reply frontend/src/components/Vote/base.vue]: I think it's cos `--theme` is a similar tone to the regular grey color, maybe try --theme-light or something in between?
Also, maybe the down arrow should be red/orange after voting

File: backend/src/models/likes.js [hunk 3]
```diff
@@ -25,9 +22,7 @@ class Likes {
         return this.db
             .query('SELECT value AS userLiked FROM likes WHERE objectType=? AND objectID=? AND userID=?',
                 [type, id, userID])
-            .then((userLiked) => {
-                return userLiked || { userLiked: 0 }
-            })
+            .then((userLiked) => ({ userLiked: userLiked || 0 }))
```
Comment (NunoDasNeves) [inline backend/src/models/likes.js]: This returns { userLiked: 0 } or { userLiked: { userLiked: 1 } }
It breaks comments, but not questions or reviews for some reason...

**Comment by DarkPurple141** [comment]
Ahh 👍

**Pull request merged**